pkg/status/aggregation/replicaset.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2025 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregation
+
+func AggregateReplicaSetStatus(statuses []map[string]interface{}) (map[string]interface{}, error) {
+	if len(statuses) == 0 {
+		return nil, nil
+	}
+
+	if len(statuses) == 1 {
+		return statuses[0], nil
+	}
+
+	aggregatedStatus := make(map[string]interface{})
+
+	aggregatedStatus["replicas"] = GetMin(statuses, "replicas")
+	aggregatedStatus["fullyLabeledReplicas"] = GetMin(statuses, "fullyLabeledReplicas")
+	aggregatedStatus["readyReplicas"] = GetMin(statuses, "readyReplicas")
+	aggregatedStatus["availableReplicas"] = GetMin(statuses, "availableReplicas")
+	aggregatedStatus["observedGeneration"] = GetMin(statuses, "observedGeneration")
+
+	aggregatedStatus["conditions"] = aggregateReplicaSetConditions(statuses)
+
+	return aggregatedStatus, nil
+}
+
+func aggregateReplicaSetConditions(statuses []map[string]interface{}) []interface{} {
```

Comment:
The behavior of this function does not match the documentation in https://docs.kubestellar.io/unreleased-development/direct/multi-wec-aggregated-status/